packages/@aws-cdk/aws-apigateway/test/authorizers/integ.request-authorizer.ts

```diff
@@ -0,0 +1,40 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import { App, Stack } from '@aws-cdk/core';
+import * as path from 'path';
+import { MockIntegration, PassthroughBehavior, RestApi } from '../../lib';
+import {RequestAuthorizer} from '../../lib/authorizers';
+
+// Against the RestApi endpoint from the stack output, run
+// `curl -s -o /dev/null -w "%{http_code}" <url>` should return 401
+// `curl -s -o /dev/null -w "%{http_code}" -H 'Authorization: deny' <url>` should return 403
+// `curl -s -o /dev/null -w "%{http_code}" -H 'Authorization: allow' <url>` should return 200
+
+const app = new App();
+const stack = new Stack(app, 'RequestAuthorizerInteg');
```

Comment:
I ran `cdk deploy --app test/authorizers/integ.request-authorizer.js` on this code and it fails to deploy.

Please make sure that the integration test sets up a correct RequestAuthorizer and that the instructions here are accurate. The instructions seem to be a copy-paste of TokenAuthorizer.

I would expect the example to use at least two different kinds of `identitySources` and validate that both are picked up correctly.

The same applies for any other integ test you have.

Reply:
Can you include the error you received in your comment?

Reply:
Here you go -

```sh
RequestAuthorizerInteg: creating CloudFormation changeset...
  0/12 | 8:08:18 AM | CREATE_IN_PROGRESS   | AWS::CloudFormation::Stack  | RequestAuthorizerInteg User Initiated
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::IAM::Role              | MyAuthorizerFunction/ServiceRole (MyAuthorizerFunctionServiceRole8A34C19E)
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::IAM::Role              | MyRestApi/CloudWatchRole (MyRestApiCloudWatchRoleD4042E8E)
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::ApiGateway::RestApi    | MyRestApi (MyRestApi2D1F47A9)
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::CDK::Metadata          | CDKMetadata
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::IAM::Role              | MyAuthorizerFunction/ServiceRole (MyAuthorizerFunctionServiceRole8A34C19E) Resource creation Initiated
  0/12 | 8:08:40 AM | CREATE_IN_PROGRESS   | AWS::IAM::Role              | MyRestApi/CloudWatchRole (MyRestApiCloudWatchRoleD4042E8E) Resource creation Initiated
  0/12 | 8:08:41 AM | CREATE_IN_PROGRESS   | AWS::ApiGateway::RestApi    | MyRestApi (MyRestApi2D1F47A9) Resource creation Initiated
  1/12 | 8:08:41 AM | CREATE_COMPLETE      | AWS::ApiGateway::RestApi    | MyRestApi (MyRestApi2D1F47A9)
  1/12 | 8:08:42 AM | CREATE_IN_PROGRESS   | AWS::CDK::Metadata          | CDKMetadata Resource creation Initiated
  2/12 | 8:08:42 AM | CREATE_COMPLETE      | AWS::CDK::Metadata          | CDKMetadata
  3/12 | 8:08:56 AM | CREATE_COMPLETE      | AWS::IAM::Role              | MyRestApi/CloudWatchRole (MyRestApiCloudWatchRoleD4042E8E)
  4/12 | 8:08:56 AM | CREATE_COMPLETE      | AWS::IAM::Role              | MyAuthorizerFunction/ServiceRole (MyAuthorizerFunctionServiceRole8A34C19E)
  4/12 | 8:08:58 AM | CREATE_IN_PROGRESS   | AWS::ApiGateway::Account    | MyRestApi/Account (MyRestApiAccount2FB6DB7A)
  4/12 | 8:08:58 AM | CREATE_IN_PROGRESS   | AWS::Lambda::Function       | MyAuthorizerFunction (MyAuthorizerFunction70F1223E)
  4/12 | 8:08:58 AM | CREATE_IN_PROGRESS   | AWS::ApiGateway::Account    | MyRestApi/Account (MyRestApiAccount2FB6DB7A) Resource creation Initiated
  4/12 | 8:08:58 AM | CREATE_IN_PROGRESS   | AWS::Lambda::Function       | MyAuthorizerFunction (MyAuthorizerFunction70F1223E) Resource creation Initiated
  5/12 | 8:08:58 AM | CREATE_COMPLETE      | AWS::ApiGateway::Account    | MyRestApi/Account (MyRestApiAccount2FB6DB7A)
  6/12 | 8:08:59 AM | CREATE_COMPLETE      | AWS::Lambda::Function       | MyAuthorizerFunction (MyAuthorizerFunction70F1223E)
  6/12 | 8:09:01 AM | CREATE_IN_PROGRESS   | AWS::ApiGateway::Authorizer | MyAuthorizer (MyAuthorizer6575980E)
  7/12 | 8:09:01 AM | CREATE_FAILED        | AWS::ApiGateway::Authorizer | MyAuthorizer (MyAuthorizer6575980E) Invalid request input (Service: AmazonApiGateway; Status Code: 400; Error Code: BadRequestException; Request ID: 5c972aa1-097f-40ed-9414-8610c709e4dd)
	RequestAuthorizer.getResource (/Users/nija/workplace/cdk/aws-cdk/packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.js:93:16)
	\_ new LambdaAuthorizer (/Users/nija/workplace/cdk/aws-cdk/packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.js:14:34)
	\_ new RequestAuthorizer (/Users/nija/workplace/cdk/aws-cdk/packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.js:85:9)
	\_ Object.<anonymous> (/Users/nija/workplace/cdk/aws-cdk/packages/@aws-cdk/aws-apigateway/test/authorizers/integ.request-authorizer.js:20:20)
	\_ Module._compile (internal/modules/cjs/loader.js:776:30)
	\_ Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
	\_ Module.load (internal/modules/cjs/loader.js:653:32)
	\_ tryModuleLoad (internal/modules/cjs/loader.js:593:12)
	\_ Function.Module._load (internal/modules/cjs/loader.js:585:3)
	\_ Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
	\_ startup (internal/bootstrap/node.js:283:19)
	\_ bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
  7/12 | 8:09:02 AM | ROLLBACK_IN_PROGRESS | AWS::CloudFormation::Stack  | RequestAuthorizerInteg The following resource(s) failed to create: [MyAuthorizer6575980E]. . Rollback requested by user.
  7/12 | 8:09:08 AM | DELETE_IN_PROGRESS   | AWS::CDK::Metadata          | CDKMetadata
  8/12 | 8:09:08 AM | DELETE_COMPLETE      | AWS::ApiGateway::Authorizer | MyAuthorizer (MyAuthorizer6575980E)
```

You should be able to run this yourself (might have to use the `cdk` binary from the checked out repo instead of the globally installed one) and as mentioned, ensure that all integration tests are valid cdk apps that can be deployed.